features/build_runner/build_runner.rb

```diff
@@ -1,11 +1,190 @@
+require_relative "../shared/models/artifact"
+
 module FastlaneCI
-  # Abstract class that represents a build runner, used
+  # Class that represents a BuildRunner, used
   # to run tests for a given commit sha
+  #
+  # Responsible for
+  # - Run the build (e.g. fastlane via FastlaneBuildRunner) and check its return status
+  # - Raise an exception if build fails, with information that can be handled by `TestRunnerService`
+  # - Reporting back a list of artifacts  to `TestRunnerService`
+  # - Measures the time of a `TestRunner`'s execution
+  # - Stores the `Build` information in version control and triggers the report of the build status on GitHub
+  # - Offer a way to subscribe to new lines being added to the output (e.g. to stream them to the user's browser)
+  #
   class BuildRunner
+    include FastlaneCI::Logging
+
+    # Reference to the FastlaneCI::Project of this particular build run
+    attr_accessor :project
+
+    # The code hosting service we want to report the status back to
+    # TODO: this probably shouldn't be stored here, think about using
```

Comment:
Update this as https://github.com/fastlane/ci/pull/230/files#r172956779 is resolved